### PR TITLE
Make GraphQLStoreQueryResolver contextual

### PR DESCRIPTION
**Pull request by devknoll**
Builds on #567. Part of #558

### CODE REVIEW

File: src/legacy/store/GraphQLStoreQueryResolver.js [hunk 1]
```diff
@@ -246,8 +243,8 @@ class GraphQLStoreSingleQueryResolver {
       if (subscribedIDs) {
         // always subscribe to the root ID
         subscribedIDs[nextID] = true;
-        this._subscription = GraphQLStoreChangeEmitter.addListenerForIDs(
-          Object.keys(subscribedIDs),
+        this._subscription = this._storeData.getChangeEmitter().addListenerForIDs(
+          Object.keys(subscribedIDs).map(id => this._getCanonicalID(id)),
```
Comment (josephsavona) [inline src/legacy/store/GraphQLStoreQueryResolver.js]: Why this change?

File: src/legacy/store/GraphQLStoreChangeEmitter.js [hunk 2]
```diff
@@ -22,11 +21,10 @@ export type ChangeSubscription = {
   remove: () => void;
 };
 
-var batchUpdate = callback => callback();
-var subscribers = [];
-
-var executingIDs = {};
-var scheduledIDs = null;
+type Subscriber = {
+  callback: Function,
```
Comment (josephsavona) [inline src/legacy/store/GraphQLStoreChangeEmitter.js]: nit: it's always better to be as detailed as possible with types, in this case the callback is `() => void`.

File: src/store/readRelayQueryData.js [hunk 3]
```diff
@@ -34,6 +34,8 @@ var callsToGraphQL = require('callsToGraphQL');
 var invariant = require('invariant');
 var validateRelayReadQuery = require('validateRelayReadQuery');
 
+var rangeUtils = RelayStoreData.getDefaultInstance().getRangeData();
```
Comment (josephsavona) [inline src/store/readRelayQueryData.js]: nit: consistent naming, how about `rangeData` to match the module name?

File: src/legacy/store/GraphQLStoreQueryResolver.js (hunk 1 above)
Comment (devknoll) [inline src/legacy/store/GraphQLStoreQueryResolver.js]: Right, so `GraphQLChangeEmitter` used to map range -> canonical ID for in `addListenerForIDs`, so moving the logic here just keeps the old behavior intact.

Comment (josephsavona) [inline src/legacy/store/GraphQLStoreQueryResolver.js]: :+1:

File: src/legacy/store/GraphQLStoreChangeEmitter.js (hunk 2 above)
Comment (devknoll) [inline src/legacy/store/GraphQLStoreChangeEmitter.js]: Good catch :+1: `() => void` could probably become its own `SubscriptionCallback` type at this point now too...

Comment (josephsavona) [inline src/legacy/store/GraphQLStoreChangeEmitter.js]: yup

File: src/legacy/store/GraphQLStoreQueryResolver.js (hunk 1 above)
Comment (devknoll) [inline src/legacy/store/GraphQLStoreQueryResolver.js]: That said, I think it might be a good idea to nix as many of these `this._storeData.get*()` calls as possible, and try to expose just the bare minimum API from `RelayStoreData`.

So this would just become `this._storeData.addListenerForIDs`, and `RelayStoreData` would take care of canonicalization and ushering the request to the change emitter.

Thoughts?

Comment (josephsavona) [inline src/legacy/store/GraphQLStoreQueryResolver.js]: Completely agreed, but let's change one thing at a time. First global -> contextual and make sure that's all good, then we can clean up the API.

**Comment by josephsavona** [comment]
@devknoll I'll be importing these internally and recreating your stack of commits to isolate each change, bear with me.

**Comment by josephsavona** [comment]
@facebook-github-bot import

**Comment by josephsavona** [comment]
merging this with #565
